services/motion/builtin/builtin.go

```diff
@@ -452,3 +334,172 @@ func (ms *builtIn) PlanHistory(
 	defer ms.mu.RUnlock()
 	return ms.state.PlanHistory(req)
 }
+
+func (ms *builtIn) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	ms.mu.RLock()
+	defer ms.mu.RUnlock()
+	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
+
+	resp := make(map[string]interface{}, 0)
+	if req, ok := cmd[DoPlan]; ok {
+		bytes, err := json.Marshal(req)
+		if err != nil {
+			return nil, err
+		}
+		var moveReqProto pb.MoveRequest
+		err = json.Unmarshal(bytes, &moveReqProto)
+		if err != nil {
+			return nil, err
+		}
+		moveReq, err := motion.MoveReqFromProto(&moveReqProto)
+		if err != nil {
+			return nil, err
+		}
+		plan, err := ms.plan(ctx, moveReq)
+		if err != nil {
+			return nil, err
+		}
+		resp[DoPlan] = plan.Trajectory()
+	}
+	if req, ok := cmd[DoExecute]; ok {
+		var trajectory motionplan.Trajectory
+		if err := mapstructure.Decode(req, &trajectory); err != nil {
+			return nil, err
+		}
+		if err := ms.execute(ctx, trajectory); err != nil {
+			return nil, err
+		}
+		resp[DoExecute] = true
+	}
+	return resp, nil
+}
+
+func (ms *builtIn) plan(ctx context.Context, req motion.MoveReq) (motionplan.Plan, error) {
```

Comment:
do we need to make sure the the request is formatted correctly?
Suppose the user forgot to specify a Destination, does that means that `req.Destination == nil`?